pkg/skaffold/runner/new.go

```diff
@@ -260,7 +260,7 @@ func isKptRendererOrDeployerUsed(pipelines runcontext.Pipelines) bool {
 		renderConfig := pipeline.Render
 		deployConfig := pipeline.Deploy
 
-		if renderConfig.Validate != nil || renderConfig.Transform != nil || renderConfig.Kpt != nil {
+		if renderConfig.Kpt != nil {
```

Comment:
Why are these checks different now?

Reply:
> Why is this conditional different now?

yeah.. the function was mainly used to determine if a kpt hydrationDir should be created , because the original implementation around transform/validate is to marshal related config into a kptfile and put the kptfile and intermediary manifests  into the hydrationDir and then call ``kpt render`` to do the actual transformation in-place in that direcotry,  then call ``kpt source`` to get the result into memory.  In the new flow, we run krm function against a manifest input stream with ``kpt fn eval``, we don't create any directories.  Also, ``kpt`` here for renderConfig.Transform and renderConfig.Validate  is a krm function executor, this can be replace by internal function or ``kustomize run fn``. This is not tied to ``kpt render`` or ``kpt live apply`` , as the method name suggested ``isKptRendererOrDeployerUsed``,  those conditions shouldn't be here.